src/frontend/src/pages/MyGroup/PendingApplications.jsx

```diff
@@ -1,33 +1,93 @@
-import { Table, TableBody, TableContainer } from '@carbon/react';
-
-import { SmsDataTable, SimpleHeader, SimpleRow } from 'components/SmsDataTable';
+import {
+    Table,
+    TableBody,
+    TableContainer,
+    TableCell,
+    TableRow,
+    TableBatchActions,
+    TableBatchAction,
+    TableToolbar,
+    TableSelectRow,
+    TableHeader,
+    TableHead,
+    TableSelectAll
+} from '@carbon/react';
+import { Close } from '@carbon/react/icons';
+import { SmsDataTable } from 'components/SmsDataTable';
+import { useManageGroupApplication } from 'hooks';
 
 export function PendingApplications(groupApplications) {
+    const { rejectGroupApplication } = useManageGroupApplication();
     const headers = [
         { key: 'name', header: 'Name', isSortable: true },
         { key: 'message', header: 'Message' }
-        // TODO: add management logic
-        // { key: 'options', header: 'Options' }
     ];
     const applications = groupApplications.groupApplications;
 
+    const batchActionClick = selectedRows => () => {
+        rejectGroupApplication({
+            ids: selectedRows.filter(data => data.hasOwnProperty('id')).map(data => data.id)
+        });
+        window.location.reload(false);
+    };
+
     return (
         <SmsDataTable rows={applications} headers={headers} className='groups-page__datatable'>
-            {({ rows, headers, getHeaderProps, getTableProps }) => (
-                <TableContainer
-                    title='Pending Group Application'
-                    description='This is a list of all the group application for this semester.'
-                >
-                    <Table {...getTableProps()}>
-                        <SimpleHeader headers={headers} getHeaderProps={getHeaderProps} />
-                        <TableBody>
-                            {rows.map(row => (
-                                <SimpleRow key={row.id} row={row} />
-                            ))}
-                        </TableBody>
-                    </Table>
-                </TableContainer>
-            )}
+            {({
+                rows,
+                headers,
+                getHeaderProps,
+                getTableProps,
+                getSelectionProps,
+                getRowProps,
+                getToolbarProps,
+                selectedRows,
+                getBatchActionProps
+            }) => {
+                const batchActionProps = getBatchActionProps();
+
+                return (
+                    <TableContainer
+                        title='Pending Group Application'
+                        description='This is a list of all the group application for this semester.'
+                    >
+                        <TableToolbar {...getToolbarProps()}>
```

Comment:
this toolbar has a visual glitch, courtesy of CBS,

![image](https://user-images.githubusercontent.com/26985949/235302231-28c36efb-0dac-4c96-8598-9c645053bd78.png)

we'll need an overide for `cds--actions-list` (a flex container) to include `column-gap: 4px`. This should be just enough to stop the buttons bumping into each other

Reply:
The screenshot shows half the story, this is more apparent as the on-hover effects kick-in.